Add API key prefix support

kleest opened the pull request:
This adds support for API key prefixes to assist secret scanning. The default API key prefix is "dta_" and may be changed with the alpine.secret.api.keyprefix property.

## Addressed Issue
See https://github.com/DependencyTrack/dependency-track/issues/2543

## Additional Details
This does not invalidate old API keys that do not use the new format. The user is free to entirely disable the prefix by setting the property to an empty string.

nscuro commented on alpine-common/src/main/java/alpine/Config.java:
```diff
@@ -101,6 +101,7 @@ public enum AlpineKey implements Key {
         WORKER_THREAD_MULTIPLIER               ("alpine.worker.thread.multiplier",   4),
         DATA_DIRECTORY                         ("alpine.data.directory",             "~/.alpine"),
         SECRET_KEY_PATH                        ("alpine.secret.key.path",            null),
+        SECRET_API_KEYPREFIX                   ("alpine.secret.api.keyprefix",       "dta_"),
```
I'd suggest to rename this to something like `alpine.api.key.prefix` to avoid confusion with the secret key, which is used for encryption.

nscuro replied:
While DT is the main adopter of Alpine, it's technically not the only app using it. So the default prefix should not be specific to DT (`dta`). I'd suggest either leaving it empty or using a generic `alpine` prefix. This propertly will need to be overridden in DT's `application.properties`.

nscuro reviewed (changes requested):
Thanks for the PR @kleest!

Few comments. Also, the key masking logic here

https://github.com/stevespringett/Alpine/blob/7615930ec767370cc6bd63a7f25f4456d8874752/alpine-model/src/main/java/alpine/model/ApiKey.java#L98

needs to be updated to acommodate for prefixed API keys. It currently assumes a static length of 32 characters. 

You can likely just re-use the logic from https://github.com/stevespringett/Alpine/commit/7b79d39565020738afc2310764024a2a9217440b (which fixed GHSA-gh7v-4hxp-gqp4). It will mask all characters of the API key except the last four.

kleest commented:
@nscuro Sorry for the late reply. I updated the Config class with your suggestions and rebased on master.

Furthermore, I refactored the API key masking logic. Please have a look if this may lead to unintended consequences. The API key prefix will not be masked but only if the API key has the currently configured prefix. This is for backwards compatibility to handle older API keys without or a different prefix.

I was struggling a bit with the usage of `ApiKey.getName()`. Is it a requirement that the Principal username does never change (e.g. interaction with external systems)? If so, I have to change the logic. In the new implementation, it will change if the API key prefix configuration value changes. This also means that every existing API key will have a different value for `ApiKey.getName()`.

stevespringett commented:
@nscuro any other feedback on this PR?

@kleest could you please resolve the conflicts so this can be merged?

kleest commented:
@stevespringett Done.

nscuro reviewed (approved):
Looks good; Tested with DT and both the old un-prefixed keys and the new ones with prefix work as expected. Masking in logs also works as expected. Thanks @kleest!

@stevespringett This is good to go from my side.
